assets/js/slackbot.js

```diff
@@ -0,0 +1,274 @@
+      channel: channel,
+      blocks: [
+        {
+          type: "section",
+          text: {
+            type: "mrkdwn",
+            text: "*Reminder to submit lunch order before 1pm today*:alarm:",
+          },
+        },
+        {
+          type: "actions",
+          elements: [
+            {
+              type: "button",
+              text: {
+                type: "plain_text",
+                text: "Lunch Link :sandwich:",
+                emoji: true,
+              },
+              value: "lunch_link",
+              url: linkURL,
+              action_id: "messageLinkedClicked",
+            },
+          ],
+        },
+      ],
+    });
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function listScheduledMessages() {
+  try {
+    const result = await app.client.chat.scheduledMessages.list({
+      token: process.env.SLACK_BOT_TOKEN,
+    });
+
+    for (const message of result.scheduled_messages) {
+      console.log(message);
+    }
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function deleteAllScheduledMessages() {
+  try {
+    const result = await app.client.chat.scheduledMessages.list({
+      token: process.env.SLACK_BOT_TOKEN,
+    });
+
+    for (const message of result.scheduled_messages) {
+      await app.client.chat.deleteScheduledMessage({
+        token: process.env.SLACK_BOT_TOKEN,
+        channel: message.channel_id,
+        scheduled_message_id: message.id,
+      });
+    }
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function numberOfScheduledMessages() {
+  try {
+    const result = await app.client.chat.scheduledMessages.list({
+      token: process.env.SLACK_BOT_TOKEN,
+    });
+
+    var x = 0;
+    for (const message of result.scheduled_messages) {
+      x += 1;
+    }
+    console.log(x);
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+async function scheduleWeeklyMessages() {
+  try {
+    await deleteAllScheduledMessages();
+    const nextMonday = new Date();
+    nextMonday.setDate(
+      nextMonday.getDate() + ((((7 - nextMonday.getDay()) % 7) + 1) % 7)
+    );
+
+    const nextWednesday = new Date();
+    nextWednesday.setDate(
+      nextWednesday.getDate() + ((((7 - nextWednesday.getDay()) % 7) + 3) % 7)
+    );
+
+    nextMonday.setHours(18, 0, 0);
+    await scheduleLunchBlock(TestChannel, BLDRLunchLink, nextMonday.valueOf());
+    nextMonday.setHours(16, 0, 0);
+    await scheduleLunchBlock(TestChannel, PVDLunchLink, nextMonday.valueOf());
+
+    nextWednesday.setHours(18, 0, 0);
+    await scheduleLunchBlock(
+      TestChannel,
+      BLDRLunchLink,
+      nextWednesday.valueOf()
+    );
+    nextWednesday.setHours(16, 0, 0);
+    await scheduleLunchBlock(
+      TestChannel,
+      PVDLunchLink,
+      nextWednesday.valueOf()
+    );
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+(async () => {
+  await app.start(process.env.PORT || 3000);
+  scheduleWeeklyMessages();
```

Comment:
We need to write either a Cron job in `config.exs` for calling this function once a week. Maybe look into importing the function from this .js file (something like [this](https://hexdocs.pm/elixir_script/javascriptinterop.html)), and then have the Cron job go there. We can also somehow schedule this by using the Date class with Javascript and using intervals, although I feel like a Cron job would be

Reply:
I wonder if it's possible to add a recurring reminder for a channel via the api?  Then it only needs to be done once & slack takes care of the rest.  I see there is a field for extra args in the tester, but am not sure whether #channels are supported in the api.  It is possible to do manually.